src/main/java/com/microfocus/octane/plugins/views/CoverageResource.java

```diff
@@ -47,12 +52,49 @@ public CoverageResource(UserManager userManager) {
     }
 
     @GET
-    public Response getCoverage(@Context HttpServletRequest request, @QueryParam("project-key") String projectKey, @QueryParam("issue-key") String issueKey, @QueryParam("issue-id") String issueId) {
+    public Response getCoverage(@Context HttpServletRequest request, @QueryParam("project-key") String projectKey, @QueryParam("issue-key") String issueKey, @QueryParam("issue-id") String issueId,
+                                @QueryParam("workspace-config-id") String workspaceConfigId, @QueryParam("workspace-id") String workspaceId) {
         UserProfile userProfile = userManager.getRemoteUser(request);
         if (userProfile == null) {
             return Response.status(Response.Status.UNAUTHORIZED).build();
         }
-        Map<String, Object> contextMap = CoverageUiHelper.buildCoverageContextMap(projectKey, issueKey, issueId);
+        Map<String, Object> contextMap = CoverageUiHelper.buildCoverageContextMap(projectKey, issueKey, issueId, workspaceConfigId, workspaceId);
         return Response.ok(contextMap).build();
     }
+
+    @GET
+    @Path("/octane-workspaces")
+    public Response getOctaneWorkspaces(@Context HttpServletRequest request, @QueryParam("project-key") String projectKey, @QueryParam("issue-type") String issueType) {
+        UserProfile userProfile = userManager.getRemoteUser(request);
+        if (userProfile == null) {
+            return Response.status(Response.Status.UNAUTHORIZED).build();
+        }
+
+        List<WorkspaceConfiguration> workspaceConfigsForProjectKey = ConfigurationManager.getInstance().getWorkspaceConfigurations().stream()
+                .filter(wsc -> workspaceConfigContainsProjectAndIssueType(projectKey, issueType, wsc))
+                .collect(Collectors.toList());
+
+        List<Map<String, String>> wsConfigsMap = workspaceConfigsForProjectKey.stream()
+                .map(this::toMap)
+                .collect(Collectors.toList());
+
+        return Response.ok(wsConfigsMap).build();
+    }
+
+    private static boolean workspaceConfigContainsProjectAndIssueType(String projectKey, String issueType, WorkspaceConfiguration workspaceConfiguration) {
+        return workspaceConfiguration.getJiraProjects().stream().anyMatch(jiraProject -> jiraProject.equals(projectKey))
+                && workspaceConfiguration.getJiraIssueTypes().stream().anyMatch(jiraIssueType -> jiraIssueType.equals(issueType));
+    }
+
+    private Map<String, String> toMap(WorkspaceConfiguration wsc) {
+        Map<String, String> map = new HashMap<>();
+
+        map.put("id", wsc.getId());
+        map.put("spaceConfigId", wsc.getSpaceConfigurationId());
+        map.put("spaceConfigName", ConfigurationManager.getInstance().getSpaceConfigurationById(wsc.getSpaceConfigurationId(), false).get().getName());
+        map.put("workspaceId", String.valueOf(wsc.getWorkspaceId()));
+        map.put("workspaceName", wsc.getWorkspaceName());
```

Comment:
Don't you have these in some generic consts?
If not. might be good to create such file now